pkg/daemon/rpm-ostree.go

```diff
@@ -185,11 +193,176 @@ func (r *RpmOstreeClient) RebaseLayered(imgURL string) error {
+	// from local storage -> Add a specific rule to allow the image
+	if !containerStoragePoliciesPresent {
+		policy.Transports[imagePolicyTransportContainerStorage] = make(map[string]signature.PolicyRequirements)
+	}
+	policy.Transports[imagePolicyTransportContainerStorage][url] = signature.PolicyRequirements{
+		signature.NewPRInsecureAcceptAnything(),
+	}
+
+	// Prepare the json patched content
+	policyJSON, err := json.MarshalIndent(policy, "", "  ")
+	if err != nil {
+		return err
+	}
+
+	// The temporal policy is written atomically
+	if err = writeFileAtomicallyWithDefaults(rpmOstreeTemporalPolicyFile, policyJSON); err != nil {
+		return err
+	}
+
+	if err = writeTemporalOstreePolicyFileDropin(); err != nil {
+		return err
+	}
+
+	klog.Infof("Temporal allow policy added for URL %s", url)
+	return nil
+}
+
+// generateTransportPolicyKeyForReference creates the reference string used as a key in the
+// image policy file for the "containers-storage" transport.
+//
+// The format of this string is not officially documented and was determined by reverse-engineering
+// the policy evaluation logic. It is structured as follows:
+// "[<storage-driver>@<graph-root>]<repo-digest>@<image-id>"
+//
+// The storage driver and graph root details are retrieved from the running Podman
+// instance at runtime.
+func generateTransportPolicyKeyForReference(podmanImageInfo *PodmanImageInfo) (string, error) {
+	podmanInfo, err := GetPodmanInfo()
+	if err != nil {
+		return "", fmt.Errorf("failed to get podman info for storage configuration gathering: %w", err)
+	}
+	return fmt.Sprintf("[%s@%s]%s@%s", podmanInfo.Store.GraphDriverName, podmanInfo.Store.GraphRoot, podmanImageInfo.RepoDigest, podmanImageInfo.Id), nil
+}
+
+// writeTemporalOstreePolicyFileDropin creates a systemd drop-in configuration that
+// bind-mounts the temporary policy file over the actual policy file for rpm-ostreed.
+// This allows rpm-ostree to use the patched policy without modifying the system's
+// policy file directly. After writing the drop-in, the function reloads systemd and
+// restarts rpm-ostreed to apply the changes.
+func writeTemporalOstreePolicyFileDropin() error {
+	// Create a temporal dropin to mount the temporal policy into rpm-ostreed process
+	if err := writeFileAtomicallyWithDefaults(
+		rpmOstreeTemporalDropinFile,
+		[]byte(
+			fmt.Sprintf(
+				"[Service]\nBindReadOnlyPaths=%s:%s", rpmOstreeTemporalPolicyFile, imagePolicyFilePath),
+		),
+	); err != nil {
+		return err
+	}
+	return systemdRpmOstreeReload()
+}
+
+// cleanupTemporalOstreePolicyFileDropin removes the systemd drop-in configuration
+// created by writeTemporalOstreePolicyFileDropin, restoring rpm-ostreed to use the
+// original system policy file. After removing the drop-in, the function reloads
+// systemd and restarts rpm-ostreed to apply the changes.
+func cleanupTemporalOstreePolicyFileDropin() error {
+	if err := os.Remove(rpmOstreeTemporalDropinFile); err != nil {
```

Comment:
I suppose since we're using the `/run` location now, technically any rpm-ostree rebase will queue a reboot and remove it. Fine to keep around but I guess not 100% necessary.